File: Data Structures & Algorithms/anagram-groups/submission-1.go
func groupAnagrams(strs []string) [][]string {
    n := len(strs)
    res := [][]string{}
    visited := make([]bool, n)

    for i:=0;i<n;i++{
        if visited[i]{
            continue
        }

        group := []string{strs[i]}
        visited[i] = true

        for j:=i+1;j<n;j++{
            if !visited[j] && isAnagram(strs[i], strs[j]){
                group = append(group, strs[j])
                visited[j] = true
            }
        }
        res = append(res, group)
    }
    return res
}

func isAnagram(s string, t string) bool {
    if len(s) != len(t){
        return false 
    }

    sArr := []byte(s)
    tArr := []byte(t)

    sort.Slice(sArr, func(i, j int) bool { return sArr[i] < sArr[j] })
    sort.Slice(tArr, func(i,j int) bool { return tArr[i] < tArr[j]})

    for i := range sArr{
        if sArr[i] != tArr[i]{
            return false
        }
    }

    return true
}

File: Data Structures & Algorithms/path-crossing/submission-1.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        visited = set()
        x, y = 0, 0
        for i in range(len(path)):
            visited.add((x,y))
            
            if path[i] == 'N':
                y += 1
            elif path[i] == 'E':
                x += 1
            elif path[i] == 'W':
                x -= 1
            elif path[i] == 'S':
                y -= 1
            
            if (x,y) in visited:
                return True
        
        return False
